### PR TITLE
28356-storage

### PR DESCRIPTION
**Pull request by mmiddleton3301**

### CODE REVIEW

File: src/Dfe.Spi.RatesAdapter.Application/RatesManager.cs [hunk 1]
```diff
@@ -1,21 +1,147 @@
...
         {
-            throw new System.NotImplementedException();
+            Rates toReturn = null;
+
+            if (entityName == nameof(LearningProvider))
+            {
+                long urn;
+                if (!long.TryParse(identifier, out urn))
+                {
+                    // TODO: Throw some sort of exception.
```
Comment (simonbu11) [inline src/Dfe.Spi.RatesAdapter.Application/RatesManager.cs:47]: Probably need this one

File: src/Dfe.Spi.RatesAdapter.Application/RatesManager.cs [hunk 2]
```diff
@@ -1,21 +1,147 @@
...
+                        urn,
+                        cancellationToken)
+                        .ConfigureAwait(false);
+
+                toReturn = Map(schoolInformation);
+            }
+            else
+            {
+                throw new NotImplementedException(
+                    "Not yet done! Implement me!");
```
Comment (simonbu11) [inline src/Dfe.Spi.RatesAdapter.Application/RatesManager.cs:62]: While in here, probably make this something more obvious, like "Rates for anything but LearningProvider is not supported yet". Just so we know what it is if we get it in the logs

File: src/Dfe.Spi.RatesAdapter.FunctionApp/Functions/GetRates.cs [hunk 3]
```diff
@@ -67,27 +67,45 @@ public class GetRates
...
+                new char[] { '-' },
+                StringSplitOptions.RemoveEmptyEntries);
+
+            if (idParts.Length != 3)
             {
-                rates = await this.ratesManager.GetRatesAsync(
-                    id,
-                    cancellationToken)
-                    .ConfigureAwait(false);
+                // TODO: Throw some sort of exception.
```
Comment (simonbu11) [inline src/Dfe.Spi.RatesAdapter.FunctionApp/Functions/GetRates.cs:85]: Doing this one feels important too

File: src/Dfe.Spi.RatesAdapter.FunctionApp/Functions/GetRates.cs [hunk 4]
```diff
@@ -67,27 +67,45 @@ public class GetRates
...
+
+            int year;
+            if (!int.TryParse(yearStr, out year))
             {
-                // TODO: Remove when we're all implemented.
-                rates = new Rates()
-                {
-                    Name = "Rates for Learning Provider/LA",
-                };
+                // TODO: Throw an exception.
```
Comment (simonbu11) [inline src/Dfe.Spi.RatesAdapter.FunctionApp/Functions/GetRates.cs:93]: Probably want an error here too

File: src/Dfe.Spi.RatesAdapter.Application/RatesManager.cs [hunk 5]
```diff
@@ -1,21 +1,147 @@
...
+            }
+
+            return toReturn;
+        }
+
+        private static Rates Map(
```
Comment (simonbu11) [inline src/Dfe.Spi.RatesAdapter.Application/RatesManager.cs:68]: As you have AutoMapper already referenced in the solution, would it not be easier to use it here too?

**Review by simonbu11 — APPROVED** [review]

File: src/Dfe.Spi.RatesAdapter.Application/RatesManager.cs (hunk 1 above)
Comment (mmiddleton3301) [reply src/Dfe.Spi.RatesAdapter.Application/RatesManager.cs:47]: Going to go through today and do this as part of a tidy up exercise.

File: src/Dfe.Spi.RatesAdapter.Application/RatesManager.cs (hunk 2 above)
Comment (mmiddleton3301) [reply src/Dfe.Spi.RatesAdapter.Application/RatesManager.cs:62]: Sounds good, going to polish it up today 👍

File: src/Dfe.Spi.RatesAdapter.FunctionApp/Functions/GetRates.cs (hunk 3 above)
Comment (mmiddleton3301) [reply src/Dfe.Spi.RatesAdapter.FunctionApp/Functions/GetRates.cs:85]: Agreed!

File: src/Dfe.Spi.RatesAdapter.FunctionApp/Functions/GetRates.cs (hunk 4 above)
Comment (mmiddleton3301) [reply src/Dfe.Spi.RatesAdapter.FunctionApp/Functions/GetRates.cs:93]: Indeed!

File: src/Dfe.Spi.RatesAdapter.Application/RatesManager.cs (hunk 5 above)
Comment (mmiddleton3301) [reply src/Dfe.Spi.RatesAdapter.Application/RatesManager.cs:68]: The reason I've not is, the spi-models lib doesn't implement an interface (unlike in the infrastructure/domain), so they're not guarenteed to be the same.

This at least lets us know at compilation time after upgrading the package that something isn't mapping correctly. Looks messy, I know, but it'll be easier to manage!

**Pull request merged**